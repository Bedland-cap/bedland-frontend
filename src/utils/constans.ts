export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
