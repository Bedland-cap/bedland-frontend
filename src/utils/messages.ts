const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
