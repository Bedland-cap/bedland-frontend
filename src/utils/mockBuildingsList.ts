import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
