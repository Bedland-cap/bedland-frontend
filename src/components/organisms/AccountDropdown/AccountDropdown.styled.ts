import styled from 'styled-components';
import { HEADER_HEIGHT } from 'utils/constans';

export const DropdownBox = styled.div`
  display: block;
  height: 100%;
  position: relative;
`;

export const DropdownMenu = styled.div`
  background-color: white;
  border-radius: 0 0 1rem 1rem;
  display: flex;
  flex-direction: column;
  gap: 1rem;
  padding: 1rem;
  position: absolute;
  right: 1.5rem;
  transition: height 2s, transform 2s;
  width: 12.5rem;
  zindex: 1;
`;

export const DropdownBreak = styled.hr`
  border-bottom: 1px solid #d9d9d9;
  margin-top: 0.5rem;
`;

export const UserInfoBox = styled.div`
  align-items: center;
  cursor: pointer;
  display: flex;
  flex-direction: row;
  gap: 1rem;
  height: ${HEADER_HEIGHT}px;
  margin-right: 2rem;
`;
