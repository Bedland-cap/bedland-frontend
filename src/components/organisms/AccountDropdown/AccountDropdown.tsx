import { useState, useEffect, useRef, useContext } from 'react';
import { useAppSelector } from 'store/hooks';
import { useLogoutMutation } from 'services/user/userApi';
import AccountInfo from 'components/atoms/AccountInfo/AccountInfo';
import routes from 'App/routing/routes';
import Avatar from 'assets/img/Avatar.svg';
import IconButton from 'components/atoms/IconButton/IconButton';
import ProfileImage from 'components/atoms/ProfileImage/ProfileImage';
import DropdownItem from 'components/molecules/DropdownItem/DropdownItem';
import { useNavigate } from 'react-router-dom';
import { ThemeContext } from 'theme/ThemeContext';
import * as Styled from './AccountDropdown.styled';

const AccountDropdown = () => {
  const { palette } = useContext(ThemeContext);

  const userRole = useAppSelector((state) => state.user.role);
  const userLogin = useAppSelector((state) => state.user.login);
  const navigate = useNavigate();
  const [logout] = useLogoutMutation();
  const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false);

  const ref = useRef<HTMLDivElement | null>(null);
  const onClickOutside = () => {
    setIsDropdownOpen(false);
  };

  const handleDropdown = (): void => {
    setIsDropdownOpen(!isDropdownOpen);
  };

  const handleLogout = async (): Promise<void> => {
    const currentUser = {
      login: userLogin,
      role: userRole,
    };

    try {
      const data = await logout(currentUser);
      if (data) {
        navigate(
          currentUser.role === 'manager'
            ? routes.homeForNotLoggedInManager
            : routes.homeForNotLoggedIn,
        );
      }
    } catch (err) {
      // eslint-disable-next-line no-console
      console.log(err);
    }
  };

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent): void => {
      if (ref.current && !ref.current.contains(event.target as Node)) {
        onClickOutside();
      }
    };
    document.addEventListener('click', handleClickOutside, true);
    return () => {
      document.removeEventListener('click', handleClickOutside, true);
    };
  }, []);

  return (
    <Styled.DropdownBox ref={ref}>
      <Styled.UserInfoBox onClick={handleDropdown} data-testid='dropdown'>
        <ProfileImage size={2} src={Avatar} />
        <AccountInfo userRole={userRole} userLogin={userLogin ?? ''} />
        <IconButton icon='more' color='text' size={20} onClick={handleDropdown} />
      </Styled.UserInfoBox>

      {isDropdownOpen ? (
        <Styled.DropdownMenu data-testid='dropdown-open' palette={palette}>
          <DropdownItem
            linkRoute={routes.account}
            iconName='user'
            label='My account'
            color='inputGrey'
          />
          <DropdownItem
            linkRoute={routes.settings}
            iconName='settings'
            label='Settings'
            color='inputGrey'
          />
          <Styled.DropdownBreak />
          <DropdownItem
            linkRoute={
              userRole === 'manager' ? routes.homeForNotLoggedInManager : routes.homeForNotLoggedIn
            }
            iconName='logout'
            label='Log out'
            color='primaryDark'
            onClick={handleLogout}
          />
        </Styled.DropdownMenu>
      ) : null}
    </Styled.DropdownBox>
  );
};

export default AccountDropdown;
