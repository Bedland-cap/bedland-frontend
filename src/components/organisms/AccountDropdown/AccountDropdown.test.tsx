import { renderWithProviders } from 'test.utils';
import { fireEvent, screen } from '@testing-library/react';
import AccountDropdown from './AccountDropdown';

describe('AccountDropdown', () => {
  it('renders correctly', () => {
    const snapshot = renderWithProviders(<AccountDropdown />, {});

    expect(snapshot).toMatchSnapshot();
  });

  it('dropdown opens', async () => {
    renderWithProviders(<AccountDropdown />, {});
    const dropdown = screen.getByTestId('dropdown');
    await fireEvent.click(dropdown);
    const dropdownOpen = await screen.getByTestId('dropdown-open');
    expect(dropdownOpen).toBeTruthy();
  });
  it('dropdown closes when clicked outside', async () => {
    renderWithProviders(
      <div>
        <AccountDropdown /> <p data-testid='close-dropdown'>close dropdown</p>
      </div>,
      {},
    );
    const dropdown = screen.getByTestId('dropdown');
    const closeDropdown = screen.getByTestId('close-dropdown');
    await fireEvent.click(dropdown);
    await fireEvent.click(closeDropdown);
    const dropdownOpen = await screen.queryByTestId('dropdown-open');

    expect(dropdownOpen).toBeFalsy();
  });
});
