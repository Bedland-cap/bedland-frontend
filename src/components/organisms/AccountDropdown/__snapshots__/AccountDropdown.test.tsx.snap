// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDropdown Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  border-radius: 50%;
  height: 2rem;
  width: 2rem;
}

.c0 {
  display: block;
  height: 100%;
  position: relative;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 1rem;
  height: 70px;
  margin-right: 2rem;
  min-width: 11.5rem;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
          data-testid="dropdown"
        >
          <img
            alt="profile"
            class="c2"
            size="2"
          />
          <div
            class="c3"
          >
            <h5
              style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; color: rgb(27, 26, 26);"
            />
            <h5
              style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.75rem; line-height: 1.4; letter-spacing: 0; color: rgb(75, 75, 75);"
            >
              resident
            </h5>
          </div>
          <button
            class="c4"
            type="button"
          >
            <svg
              fill="none"
              height="20"
              viewBox="0 0 20 20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <mask
                id="mask-more"
              />
              <path
                d="M10 9.99251V10.0075M10 4V4.01498M10 15.985V16"
                stroke="#1B1A1A"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2.68161"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  border-radius: 50%;
  height: 2rem;
  width: 2rem;
}

.c0 {
  display: block;
  height: 100%;
  position: relative;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 1rem;
  height: 70px;
  margin-right: 2rem;
  min-width: 11.5rem;
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
        data-testid="dropdown"
      >
        <img
          alt="profile"
          class="c2"
          size="2"
        />
        <div
          class="c3"
        >
          <h5
            style="font-family: Source Sans Pro, sans-serif; font-weight: 600; font-size: 0.875rem; line-height: 1.4; letter-spacing: 0; color: rgb(27, 26, 26);"
          />
          <h5
            style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 0.75rem; line-height: 1.4; letter-spacing: 0; color: rgb(75, 75, 75);"
          >
            resident
          </h5>
        </div>
        <button
          class="c4"
          type="button"
        >
          <svg
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-more"
            />
            <path
              d="M10 9.99251V10.0075M10 4V4.01498M10 15.985V16"
              stroke="#1B1A1A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2.68161"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
